force-app/main/default/classes/OpportunityUtils_Test.cls

```diff
@@ -7,4 +7,110 @@ private class OpportunityUtils_Test
+            System.assertNotEquals(true, opp.Name.endsWith('Large'), 'Wrong opp size in name');
+            System.assert(opp.Name.endsWith('Small'), 'Wrong opp size in name');
+        }
+
+        for (Integer i = 50; i < 100; i++) 
+        {
+            opp = resultOpps[i];
+            System.assert(opp.Amount > 50000, 'Testing should be for Medium amounts');
+            System.assert(opp.Amount <= 100000, 'Testing should be for Medium amounts');
+            System.assertEquals(true, opp.Name.endsWith('Medium'), 'Wrong opp size in name');
+            System.assertNotEquals(true, opp.Name.endsWith('Small'), 'Wrong opp size in name');
+            System.assertNotEquals(true, opp.Name.endsWith('Large'), 'Wrong opp size in name');
+
+
+        }
+
+        for (Integer i = 100; i < 200; i++) 
+        {
+            opp = resultOpps[i];
+            System.assert(opp.Amount > 100000, 'Testing should be for Large amounts');
+            System.assertEquals(true, opp.Name.endsWith('Large'), 'Wrong opp size in name');
+            System.assertNotEquals(true, opp.Name.endsWith('Small'), 'Wrong opp size in name');
+            System.assertNotEquals(true, opp.Name.endsWith('Medium'), 'Wrong opp size in name');
+        }
+    }
+
+    //test boundaries
+    //0
+    //50000
+    //100000
+
+    @IsTest
+    private static void testBoundaryAmounts()
+    {
+        User salesUser = UnitTestHelper.CreateUser_inMemory('Joe', 'Bloggs', 'jbl', 'Standard User');
+        insert salesUser;
+
+        Account testAcc = UnitTestHelper.createAccount_InMemory();
+        insert testAcc;
+
+        //Creates 200 opps with amounts 1k to 200k
+        Opportunity opp0k   = UnitTestHelper.CreateOpportunity_InMemory(testAcc, 0);
+        Opportunity opp50k  = UnitTestHelper.CreateOpportunity_InMemory(testAcc, 50000);
+        Opportunity opp100k = UnitTestHelper.CreateOpportunity_InMemory(testAcc, 100000);
+
+        List<Opportunity> testBoundaryOpps = new List<Opportunity>{opp0k, opp50k, opp100k};
+
+        Test.StartTest();
+        System.runAs(salesUser)
+        {
+            OpportunityUtils oppUtils = new OpportunityUtils();
+            oppUtils.addOpportunitySizeToNameAndSave(testBoundaryOpps); 
+            //Why isn't this a static method?
+        }
+        Test.StopTest();
+
+        List<Opportunity> resultOpps = [SELECT Id, Name, Amount 
+                                        FROM Opportunity 
+                                        WHERE Id IN: testBoundaryOpps
+                                        ORDER BY Amount];
+
+        System.assertEquals(3, resultOpps.size(), 'Expected 3 test opps');
+
+        System.assertEquals(true, resultOpps[0].Name.endsWith('Small'), '0 should be small indeed!');
+        System.assertEquals(true, resultOpps[1].Name.endsWith('Small'), 'Exactly 50k should be small ');
+        System.assertEquals(true, resultOpps[02].Name.endsWith('Medium'), 'Exactly 100k should be medium ');
```

Comment:
I don't think you have all of the boundary test cases here - if the value is between 50000 and 100000 it should be medium, but with your tests the amount 99999 could be erroneously marked as 'Small'. Boundary tests would typically verify the exact boundary amount, one below and one above.